.github/workflows/build.yml

```diff
@@ -51,44 +51,66 @@ jobs:
       - uses: actions/checkout@v4
         with:
           persist-credentials: false
+
       - uses: ruby/setup-ruby@v1.153.0
         with:
           ruby-version: ${{ matrix.ruby }}
           bundler-cache: true
+
       - name: Rubocop
         run: bin/rake rubocop
+
       - name: Rspec tests
         run: bin/rake spec
+
       - name: Yard documents
         run: bin/rake yard
 
-  deploy:
-    needs: 'tests'
-    if: (github.event_name == 'push' || github.event.inputs.deploy == 'true') && github.ref == 'refs/heads/master'
+  build-deploy-docs:
+    needs: tests
+    if: >
+      contains(github.event.pull_request.labels.*.name, 'dependencies')
+      || github.event_name == 'push'
+      || github.event.inputs.deploy == 'true'
     runs-on: ubuntu-latest
     steps:
       - uses: actions/checkout@v4
         with:
           persist-credentials: false
+
       - uses: ruby/setup-ruby@v1.153.0
         with:
-          ruby-version: 2.7
+          ruby-version: 3.2
           bundler-cache: true
+
       - name: Yard documents
         run: bin/rake yard
 
       - name: Deploy pages
-        if: (github.event_name == 'push' || github.event.inputs.deploy == 'true')
+        if: github.ref == 'refs/heads/master' # only push on master branch
         uses: JamesIves/github-pages-deploy-action@v4.4.3
         with:
-          SSH: false
-          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
-          BRANCH: gh-pages
-          FOLDER: doc
-          CLEAN: true
-          CLEAN_EXCLUDE: '[".gitattributes", ".gitignore", "CNAME", "LICENSE"]'
+          branch: gh-pages
+          folder: doc
+          clean: true
+          clean-exclude: |
+            .gitattributes
+            .gitignore
+            LICENSE
+
+  # release-gem:
```

Comment:
Are these supposed to be commented out?

Reply:
Yes. I noticed while working on this PR that the commented out code for cutting a gem shouldn't be a part of the build-deploy-pages workflow so I figured I could extract it in anticipation for enabling it in a subsequent pull request.